spell/spell-reviewer-mainnet-checklist.md

```diff
@@ -170,7 +190,31 @@ Spell Actions:
     * [ ] Recipient Addresses match Doc
     * [ ] Payment Amounts match Doc
     * [ ] Follow Previous Patterns
+  * [ ] `Yank`
+    * [ ] Stream ID is matches exec
+    * [ ] `MCD_VEST_DAI` for DAI stream `yank`
+    * [ ] `MCD_VEST_MKR_TREASURY` for MKR stream `yank`
   * [ ] Ensure Recipient Addresses match `addresses_wallets.sol`
+* [ ] SubDAO Content
+  * [ ] SubDAO SubProxy spell execution
+    * [ ] SubDAO spell address matches exec doc
+    * [ ] SubDAO spell deployer is in `addresses_deployers`
+    * [ ] Executed using `ProxyLike(SUBDAO_PROXY).exec(SUBDAO_SPELL, abi.encodeWithSignature("execute()"));`
+    * [ ] Execution is NOT delegate call
+    * [ ] Gas cost will not be too high (low level call gas estimation is not done by our scripts)
+  * [ ] Maker Core (main spell) SubDAO actions (i.e. operate in Pause Proxy `DelegateCall` context)
+    * [ ] No SubDAO contract being interacted with is authed on a core contract like vat, etc. (script this eventually)
+    * [ ] SubDAO contract licensing and optimizations do not matter (not strictly)
+    * [ ] SubDAO contracts and all libraries / dependencies are verified (Blocking if not true)
+    * [ ] Upgradable SubDAO contracts
+      * [ ] Any upgradable contracts have the `PAUSE_PROXY` as their `admin` (i.e. the party that can upgrade)
+        * [ ] Any upgradable SubDAO contracts with an `admin` that is not `PAUSE_PROXY` are not authed on any core contracts (Blocking)
+    * [ ] All SubDAO content addresses (i.e. provided contract addresses or EOAs) present in the Maker Core spell are present in the exec doc and are correct. SubDAO addresses being authed or given any permissions MUST be in the exec doc. SubDAO addresses being called must be confirmed by the SubDAO spell team.
+      * [ ] IF addresses not PR'ed in by the SubDAO team (use git blame for example), SubDAO content addresses all have inline comment for provenance or source being OKed by SubDAO
```

Comment:
I'm not sure where is this requirement coming from. If SubDAO content is in the doc, why do we need this extra confirmation? How would then this confirmation look like? 

Suggestion: either remove it (preferred) or clearly specify what `OKed` means in very specific terms (eg. a comment in the PR)

Reply:
Yes sorry, this is a bit of a grey area.

Provenance and stuff are still not formalized but in the interim.

In my opinion, the following are OK:

Confirmation in Forum
Confirmation on GitHub via comment
Confirmation in Exec Sheet via comment
Discord Message saying it's OK (screenshot it)

This means 'written approval' in spirit.

Really this is a Governance Facilitator thing to handle so that's why it's a bit iffy.

This will be refined in the future, this is just for now.

Reply:
I also like this idea. 

However, can we do this at a separate time when we can get stakeholder involvement (SubDAO representatives as well as the subdao spell reviewers, which is who I assume you are referring to when you say 'the reviewers').

On the "code matches deployed address" stuff, this is outside of core spell team scope for a variety of technical reasons, as there is no infra in place like the extcodehash (soul) [check](https://github.com/dapphub/ds-pause/blob/master/src/pause.sol#L97) present for SubDAOs at present.

IMO this is a good point but should be tackled separately to avoid inflammation of what is already a sensitive issue.

Does this work for you?

Reply:
Agreed that this can be discussed and done in a separate PR